File: frontend/src/types/entities/payment-method.ts
/**
 * ============================================================================
 * Hela360 Payment Method Entity
 * ============================================================================
 *
 * Tenant-owned tender method reference data returned by the backend for POS
 * checkout payment selection.
 *
 * ============================================================================
 */

export interface PaymentMethod {
  id: string;

  code: string;

  name: string;

  method_type: string;

  is_active: boolean;
}

File: app/serializers/payment_method.py
from app.models import PaymentMethod


def serialize_payment_method(
    payment_method: PaymentMethod,
) -> dict:
    return {
        "id": payment_method.id,
        "code": payment_method.code,
        "name": payment_method.name,
        "method_type": payment_method.method_type,
        "is_active": payment_method.is_active,
    }
